### DISCUSSION TITLE
Expose core `classed` function in react package

### DISCUSSION THREAD

**Discussion by rcoopr** [sannajammeh/tw-classed#45] [Ideas]
Personally I would still find value from a function that produces only a classname, and not a component, even within a React context.

The issue with the current setup is bad DX when importing `classed` as both functions exposed by @tw-classed/react and @tw-classed/core are named the same

Some alternative solutions could be 
 - renaming the `classed` function in the react package to `component` or something similar
 - adding the `componentising` feature as an overload on the core function instead, perhaps based on the value of the first argument. 

Opening this issue as a discussion. I like the direction of this library. Reminds me of [cva](https://github.com/joe-bell/cva)

**Comment by sannajammeh** [comment]
Thanks for this! 

I agree it would be ideal to have them both work. Writing 
```js
import { classed as cn } from "@tw-classed/core";
import { classed } from "@tw-classed/react"; 
```
might feel a bit cumbersome.

I don't think the React function should change however. Classed is specifically referring to applying classes to a component. This has been the case in many usage guides writing custom `classed` functions on medium for example. 

Adding the `core` function will result in a higher bundle size of `~300b`. If this is a compromise most would make then I'm happy to expose an aliased version of `core` in the react library. As this library aims to be as close to `Stitches.js` as possible this is a great idea.

Syntax would be as follows

```js
import { classed, clc } from "@tw-classed/react"; // Example 1
import { classed, core } from "@tw-classed/react"; // Example 2

// Example 3 direct property ( Bundle size will increase with 200-300b here )
import { classed } from "@tw-classed/react";

const button = classed.core("bg-blue-500", {...variants}, "more classes");
```

I certainly prefer the `classed.core´ method, but anything goes.

Feel free to come up with any names :)
